### DISCUSSION TITLE
Entity - change material of model in code

### DISCUSSION THREAD

**Discussion by Generalec** [stride3d/stride#2158] [Q&A] [answered]
Hello,

I have imported model with few material slots.

I want in C# code change specific material when specific action happen.

_Just my idea for understanding my question_
`Entity.Material55 = "Models/MyMaterialBlue"`

I have been trying to find out something but I could not find specific way how to achieve this.

Thank you in advance.

**Comment by tebjan** [comment]
You can set the material on the model, the entity doesn't know about materials. Materials are not part of the entity-component system, they are simply properties of the model. You can load as many materials as you want and assign them to the model in code.

**Comment by Generalec** [comment]
> ....ou can load as many materials as you want and assign them to the model in code.

Can you please give me hint about function or code example, please?

Thank you.
....


[Edit: I try to dig into if this is solution.](https://doc.stride3d.net/latest/en/manual/graphics/materials/material-slots.html?q=model)

**Comment by Generalec** [comment, marked as answer]
[Material slots](https://doc.stride3d.net/latest/en/manual/graphics/materials/material-slots.html?q=model)
